src/App.tsx

```diff
@@ -1,28 +1,38 @@
 import { useAppSelector } from "hooks";
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 import { useDispatch } from "react-redux";
 import { setDarkTheme } from "appRedux/themeSlice";
-import { AppRoutes, Header } from "./components";
+import { AppRoutes, Header, Sidebar } from "./components";
 import { Link } from "react-router-dom";
-import { setToken, setUser } from "appRedux/userSlice";
-import { getUserFromToken } from "apiCalls";
+import { setToken, setUser, updateUserState } from "appRedux/userSlice";
+import { getPosts, getUserFromToken } from "apiCalls";
+import { setPosts } from "appRedux/postSlice";
 
 function App() {
   const darkTheme = useAppSelector((state) => state.theme);
+  const token = useAppSelector((state) => state.userData.encodedToken);
+  const [isInitialRender, setIsInitialRender] = useState(true);
+
   const dispatch = useDispatch();
 
   // For auto login
   useEffect(() => {
     (async () => {
       try {
-        const token = localStorage.getItem("token") ?? "";
-        if (token !== "") {
-          const user = await getUserFromToken(token).then((res) => res.data);
-          dispatch(setToken(token));
-          dispatch(setUser(user));
+        const token = localStorage.getItem("token");
+        if (token !== "" && token !== null && token !== undefined) {
+          const user = await getUserFromToken(token).then(
+            (res) => res.data.user
+          );
+
+          dispatch(updateUserState({ encodedToken: token, user }));
+          const posts = await getPosts().then((res) => res.data.posts);
+          dispatch(setPosts(posts));
         }
       } catch (err) {
         console.error(err);
+      } finally {
```

Comment:
you can do initial call in index.js intself if you are using. redux you can do store.dispatch(andyour function here)